docs/usage/notes.rst

```diff
@@ -233,10 +236,19 @@ won't free disk space, they merely tag data as deleted in a new transaction.
 Be aware that as soon as you write to the repo in non-append-only mode (e.g. prune,
 delete or create archives from an admin machine), it will remove the deleted objects
 permanently (including the ones that were already marked as deleted, but not removed,
-in append-only mode).
+in append-only mode). Automated edits to the repository (such as a cron job running
+``borg prune``) will render append-only mode moot if data is deleted.
+
+Even if an archive appears to be available, it is possible an attacker could delete
+just a few chunks from an archive and silently corrupt its data. While in append-only
+mode, this is reversible, but ``borg check`` should be run before a writing/pruning
+operation on an append-only repository to catch accidental or malicious corruption::
+
+    # run without append-only mode
+    borg check --verify-data repo && borg compact repo
```

Comment:
"run before a writing/pruning operation on an append-only", once again, "writing/pruning" in which mode, append-only or not?
Also, why should `borg check --verify-data` be run without append-only?  Isn't this read-only operation?
What `compact` has to do with this discussion?

Reply:
@ThomasWaldmann, just in case you missed this question about `borg check --verify-data`, it's still relevant.  I also now found #4660 fixed by #4714, and I'm really under impression now that

```python
# run without append-only mode
```

should have been removed as well, since `borg check --verify-data` should be read-only, isn't it?  That is wasn't the above comment applicable only because of `borg compact`, which was now removed from documentation?

Reply:
Sorry to bother you so much jumping around different discussions about `check`, `prune`, `compact`.  I'm just finally in the middle of organizing that subject area of a large backup infrastructure.

Reply:
if borg check --verify-data is successful (all looks authentic / uncorrupted), it will return rc == 0 and that will cause the righthand part to execute (borg compact).

if it looks like somebody has tampered with the repo contents (like corrupted chunks), it won't execute the borg compact.

guess the idea here was to only run compact if everything looks good.
otherwise you may want to rollback the repo to an older state and for that to work, you must not run borg compact.

Reply:
I guess my point was that, for Borg 1.2, it should be:

```shell
borg check --verify-data repo && {
  # disable append-only mode and then run
  borg compact repo
}
```

While, for Borg 1.1, it should be:

```shell
borg check --verify-data repo
```

regardless of whether append-only mode is on or off.